lib/core/utils/selector-cache.js

```diff
@@ -0,0 +1,196 @@
+import { matchesExpression } from './matches';
+import tokenList from './token-list';
+
+// since attribute names can't contain whitespace, this will be
+// a reserved list for ids so we can perform virtual id lookups
+const idsKey = ' [idsMap]';
+let selectorMap = {};
```

Comment:
Do we need to put this in the axe-core cache, so that it can be cleaned up after a run?

Reply:
I attached the selector cache to the root node so once the tree is cleaned up after a run the cache goes away as well.

Reply:
Trying to make it part of the cache interfered with other tests as creating any virtual node adds the node to the cache. The cache is attached to the tree so it at least is clean up that way, but the object will continue to be around after a run.